src/workerd/api/sockets.h

```diff
@@ -69,27 +72,48 @@ struct SocketOptions {
 struct InitData {
   jsg::Ref<ReadableStream> readable;
   jsg::Ref<WritableStream> writable;
-  IoOwn<kj::PromiseFulfillerPair<void>> closeFulfiller;
+  IoOwn<kj::PromiseFulfillerPair<bool>> closeFulfiller;
```

Comment:
Why add these new members to `InitData` instead of constructing them inside `Socket`'s constructor? It doesn't look like the caller ever does anything beyond allocating them?

Reply:
Seems cleaner to do it this way, but if you prefer I can move this to the constructor.